mlir/lib/Dialect/Rock/Tuning/GridwiseGemmParams.cpp

```diff
@@ -507,46 +544,133 @@ PopulateParamsXDL::getTuningParameters(KernelType opType, Type dataTypeA,
   return res;
 }
 
-LogicalResult PopulateParamsXDL::isValidGemm(const InitParamsXDL &param,
-                                             const GemmSize &gemmSize) const {
-  if (!(gemmSize.m % param.gemmMPerBlock == 0 &&
-        gemmSize.n % param.gemmNPerBlock == 0 &&
-        gemmSize.k % (param.gemmKPerBlock * param.gemmKPack) == 0)) {
+Attribute
+PopulateParamsXDL::getGemmParamsAttr(OpBuilder builder,
+                                     InitParamsAccel validParams) const {
+  return builder.getAttr<XdlopsGemmParamsAttr>(
+      validParams.gemmKPerBlock, validParams.gemmMPerBlock,
+      validParams.gemmNPerBlock, validParams.gemmKPack,
+      validParams.gemmMPerWave, validParams.gemmNPerWave,
+      validParams.gemmAThreadCopyMoreGemmK);
+}
+
+/// Wmma acceleration
+// clang-format off
+const InitParamsAccel
+PopulateParamsWmma::initParametersFp16[PopulateParamsWmma::nInitParametersFp16] = {
+  // M/block N/block K/block M/wave N/wave kPack forceUnroll bCopyMore
+  {128,  128,  4,  64,  64,  16,  true,  true},
+  {32,  128,  4,  32,  32,  16,  true,  true},
+  {32, 64, 4, 32, 64, 16, true, true},
+};
+
+const InitParamsAccel
+PopulateParamsWmma::initParametersForward8Bit[
+  PopulateParamsWmma::nInitParametersForward8Bit] = {
+  {128, 128, 8, 64, 64, 16, true, true},
+  {64, 64, 4, 32, 32, 16, true, true},
+  {64, 64, 8, 32, 32, 16, true, true},
+  {32, 32, 4, 16, 16, 16, true, true},
+  {32, 32, 8, 16, 16, 16, true, true},
+};
+// clang-format on
+
+LogicalResult
+PopulateParamsWmma::isValidBlockwiseGemm(const InitParamsAccel &param,
+                                         Type dataTypeA, Type dataTypeB,
+                                         StringRef arch, uint32_t blockSize) {
+
+  const int64_t waveSize = mlir::rock::lookupArchInfo(arch).waveSize;
+
+  // clang-format off
+  std::vector<std::tuple<int, int, int>> validWaveGemmSize =
+  {
+    std::make_tuple(128, 128, 2),
+    std::make_tuple(128, 64, 2),
+    std::make_tuple(64, 128, 2),
+    std::make_tuple(64, 64, 2),
+    std::make_tuple(64, 32, 2),
+    std::make_tuple(32, 64, 2),
+    std::make_tuple(32, 32, 2),
+    std::make_tuple(32, 16, 2),
+    std::make_tuple(16, 32, 2),
+    std::make_tuple(32, 32, 2),
+    std::make_tuple(64, 16, 2),
+    std::make_tuple(16, 64, 2),
+    std::make_tuple(16, 16, 2),
+  };
```

Comment:
This seems to be a copy-paste from xdlops gemm. Now will some configuration such as 16x128 ever become performant in those combinations? How does CK handle those?

Reply:
I don't think they do any sort of tuning yet (from what I see). I mean it's not too different from Xdlops. I added some configuration, but I plan on pruning it once I get down to tuning